src/containers/store/filters.svelte

```diff
@@ -1,107 +1,181 @@
 <script>
-  import {get} from 'svelte/store';
+  import { createEventDispatcher } from 'svelte';
+  import SearchBar from '@/components/common/search-bar.svelte';
   import TextField from 'ui/text-field.svelte';
   import Dropdown from 'ui/dropdown.svelte';
   import Dot from 'ui/dot.svelte';
   import Button from 'ui/button.svelte';
   import RadioGroup from 'ui/radio-group.svelte';
+  import RadioChipGroup from 'ui/radio-chip-group.svelte';
   import CheckboxGroup from 'ui/checkbox-group.svelte';
 
-  import orders from '@/constants/store/order';
-  import colors from '@/constants/store/colors';
-  import {selectedOrder, selectOrder, filters, changeSearch, changeColors, changeIsColorless, changeLowPrice, changeHighPrice, clearAllFilters} from '@/store/store';
-  import {isAuthed, user} from '@/store/user';
+  export let orderOptions;
+  export let orderLabels;
+  export let colors;
+  export let order;
+  export let orderLabel;
+  export let balance;
+  const initialOrder = order;
 
-  const showAffordable = () => {
-    let {balance} = get(user);
-    if (balance) {
-      changeLowPrice(0);
-      changeHighPrice(balance);
-    }
+  let filtering = {
+    searchQuery: '',
+    minPrice: null,
+    maxPrice: null,
+    excludedColors: [],
+    colorlessExcluded: false,
+    order: order,
   };
+  $: filtersActive = ((filtering.minPrice != null)
+                   || (filtering.maxPrice != null)
+                   || (filtering.excludedColors.length !== 0)
+                   || (filtering.colorlessExcluded));
+
+  colors.forEach(colorObject => { colorObject.checked = true; });
+  let colorlessValues = [{ value: null, checked: !filtering.colorlessExcluded }];
+
+  const dispatch = createEventDispatcher();
+
+  let searchThrottle = null;
```

Comment:
This is actually debounce and not throttle

Reply:
Technically, [this is neither](https://css-tricks.com/the-difference-between-throttling-and-debouncing/). It's just delaying the call. Idk what to best call this technique.

Reply:
I'm not sure why you say it's neither. Perhaps this demo will make it clearer: http://demo.nimius.net/debounce_throttle/

Reply:
Oh, yeah, right, sorry. Will rename